spring-cloud-contract-tools/spring-cloud-contract-gradle-plugin/src/main/groovy/org/springframework/cloud/contract/verifier/plugin/SpringCloudContractVerifierGradlePlugin.java

```diff
@@ -0,0 +1,389 @@
+			ContractVerifierExtension.ContractRepository original = extension.getContractRepository();
+
+			publishStubsToScmTask.getContractRepository().getRepositoryUrl().convention(stubs.getRepositoryUrl().orElse(original.getRepositoryUrl()));
+			publishStubsToScmTask.getContractRepository().getUsername().convention(stubs.getUsername().orElse(original.getUsername()));
+			publishStubsToScmTask.getContractRepository().getPassword().convention(stubs.getPassword().orElse(original.getPassword()));
+			publishStubsToScmTask.getContractRepository().getProxyHost().convention(stubs.getProxyHost().orElse(original.getProxyHost()));
+			publishStubsToScmTask.getContractRepository().getProxyPort().convention(stubs.getProxyPort().orElse(original.getProxyPort()));
+			publishStubsToScmTask.getContractsMode().convention(extension.getContractsMode());
+			publishStubsToScmTask.getDeleteStubsAfterTest().convention(extension.getDeleteStubsAfterTest());
+			publishStubsToScmTask.getFailOnNoContracts().convention(extension.getFailOnNoContracts());
+			publishStubsToScmTask.getContractsProperties().convention(extension.getContractsProperties());
+			publishStubsToScmTask.getStubsDir().convention(generateClientStubs.flatMap(GenerateClientStubsFromDslTask::getStubsOutputDir));
+
+			publishStubsToScmTask.dependsOn(generateClientStubs);
+		});
+	}
+
+	private TaskProvider<GenerateClientStubsFromDslTask> createAndConfigureGenerateClientStubs(ContractVerifierExtension extension,
+			TaskProvider<ContractsCopyTask> copyContracts) {
+		TaskProvider<GenerateClientStubsFromDslTask> task = project.getTasks().register(GenerateClientStubsFromDslTask.TASK_NAME, GenerateClientStubsFromDslTask.class, generateClientStubs -> {
+			generateClientStubs.setGroup(GROUP_NAME);
+			generateClientStubs.setDescription("Generate client stubs from the contracts");
+
+			generateClientStubs.getContractsDslDir().convention(copyContracts.flatMap(ContractsCopyTask::getCopiedContractsFolder));
+			generateClientStubs.getExcludedFiles().convention(extension.getExcludedFiles());
+			generateClientStubs.getExcludeBuildFolders().convention(extension.getExcludeBuildFolders());
+
+			generateClientStubs.getStubsOutputDir().convention(extension.getStubsOutputDir().dir(buildRootPath(GenerateClientStubsFromDslTask.DEFAULT_MAPPINGS_FOLDER)));
+
+			generateClientStubs.dependsOn(copyContracts);
+		});
+		return task;
+	}
+
+	private void createAndConfigureStubsJarTasks(ContractVerifierExtension extension,
+			TaskProvider<ContractsCopyTask> copyContracts,
+			TaskProvider<GenerateClientStubsFromDslTask> generateClientStubs) {
+		TaskProvider<Jar> task = stubsTask();
+		if (task != null) {
+			// How is this possible? Where can it come from?
+			// TODO: This can only happen if one of the following is true:
+			//   1) plugin was applied twice (allprojects/subprojects/standard apply)
+			//   2) Another task with the same name exists, but was not provided by this plugin
```

Comment:
Yeah that was the case. Someone could have overridden the stubs jar task so we won't override their custom one.